grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -779,11 +781,14 @@ class StockMovementService {
         List<SubstitutionItem> availableSubstitutions = getAvailableSubstitutions(location, requisitionItem.product)
         List<SubstitutionItem> substitutionItems = getSubstitutionItems(location, requisitionItem)
 
+        Map monthlyStocklistQuantities = calculateMonthlyStockListQuantity(location)
```

Comment:
This is going to be expensive if we do it for every edit page item. Try to do it outside of a loop since the monthly stocklist quantity is for all items. Otherwise write a method that takes a location and product so that you're only retrieving a single value (not a map).

Reply:
@jmiranda I've changed it.